src/crystal/system/unix/process.cr

```diff
@@ -0,0 +1,227 @@
+require "c/signal"
+require "c/stdlib"
+require "c/sys/resource"
+require "c/unistd"
+
+struct Crystal::System::Process
+  getter pid : LibC::PidT
+
+  def initialize(@pid : LibC::PidT)
+    @channel = Crystal::SignalChildHandler.wait(@pid)
+  end
+
+  def wait
+    @channel.receive
+  end
+
+  def exists?
+    !@channel.closed? && Crystal::System::Process.exists?(@pid)
+  end
+
+  def terminate
+    Crystal::System::Process.signal(@pid, LibC::SIGTERM)
+  end
+
+  def self.exit(status)
+    LibC.exit(status)
+  end
+
+  def self.pid
+    LibC.getpid
+  end
+
+  def self.pgid
+    ret = LibC.getpgid(0)
+    raise RuntimeError.from_errno("getpgid") if ret < 0
+    ret
+  end
+
+  def self.pgid(pid)
+    # Disallow users from depending on ppid(0) instead of `pgid`
+    raise RuntimeError.from_errno("getpgid", Errno::EINVAL) if pid == 0
+
+    ret = LibC.getpgid(pid)
+    raise RuntimeError.from_errno("getpgid") if ret < 0
+    ret
+  end
+
+  def self.ppid
+    LibC.getppid
+  end
+
+  def self.signal(pid, signal)
+    ret = LibC.kill(pid, signal)
+    raise RuntimeError.from_errno("kill") if ret < 0
+  end
+
+  def self.exists?(pid)
+    ret = LibC.kill(pid, 0)
+    if ret == 0
+      true
+    else
+      return false if Errno.value == Errno::ESRCH
+      raise RuntimeError.from_errno("kill")
+    end
+  end
+
+  def self.times
+    LibC.getrusage(LibC::RUSAGE_SELF, out usage)
+    LibC.getrusage(LibC::RUSAGE_CHILDREN, out child)
+
+    {
+      utime:  usage.ru_utime.tv_sec.to_f64 + usage.ru_utime.tv_usec.to_f64 / 1e6,
+      stime:  usage.ru_stime.tv_sec.to_f64 + usage.ru_stime.tv_usec.to_f64 / 1e6,
+      cutime: child.ru_utime.tv_sec.to_f64 + child.ru_utime.tv_usec.to_f64 / 1e6,
+      cstime: child.ru_stime.tv_sec.to_f64 + child.ru_stime.tv_usec.to_f64 / 1e6,
+    }
```

Comment:
Why return a custom named tuple instead of Tms directly?

Reply:
It's a circular dependency on process.cr

Reply:
`Tms` just needs to be moved. But there is no sense in having two data structures for the same purpose.

Reply:
I don't understand where exactly it would be moved.

NamedTuple should be seen as a set of arguments and their values, not necessarily a permanent structure.

Reply:
In case nothing else works, move it to `src/process/tms.cr` and require that file from here.
A named tuple is a data type and we just don't need another data type for exactly the same thing.

Reply:
OK